### PR TITLE
setup: playwright

### PR DESCRIPTION
**Pull request by uchilaka**
Implement Github Actions workflow for automated testing on the following versions of node:
- `16.x`
- `18.x`
- `latest LTS`

## Issue(s) resolved
- [ ] https://github.com/hotaydev/audit-export/issues/12

## Pre-launch Checklist

- [ ] I read the [Contribution Guide](https://github.com/hotaydev/audit-export/blob/main/.github/CONTRIBUTING.md) and followed the process for submitting PRs.
- [x] I listed at least one issue this PR fixes in the description above.
- [ ] I have tested this package in at least two versions of Node.js (one lower than the v14 and one higher than the v16)

If you need help, consider asking for advice on [Github Discussions](https://github.com/hotaydev/audit-export/discussions) page.

### CODE REVIEW

File: .github/workflows/playwright.yml [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+  pull_request:
+
+jobs:
+  test:
+    timeout-minutes: 60
+    runs-on: ubuntu-latest
+    strategy: 
+      matrix:
+        node-version: 
+          # - '12'
```
Comment (uchilaka) [inline .github/workflows/playwright.yml:15]: The commented out versions had been failing while I was working on this PR. Since the PR has been stable I haven't retried them in CI yet. I'll push up a quick commit to see if they're still failing before signing off for now

File: .github/workflows/playwright.yml [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        node-version: ${{ matrix.node-version }}
+        cache: 'npm'
+
+    - name: Set short git commit SHA
+      id: vars
+      run: |
+        calculatedSha=$(git rev-parse --short ${{ github.sha }})
+        echo "COMMIT_SHORT_SHA=$calculatedSha" >> $GITHUB_ENV
+
+    # - name: Get installed Playwright version
```
Comment (uchilaka) [inline .github/workflows/playwright.yml:37]: TODO: Clean this up

File: package.json [hunk 3]
```diff
@@ -40,9 +42,13 @@
   },
   "dependencies": {
     "ejs": "^3.1.9",
-    "fs-extra": "^11.2.0"
+    "fs-extra": "^11.2.0",
+    "globals": "^15.0.0"
   },
   "devDependencies": {
+    "@eslint/eslintrc": "^3.0.2",
```
Comment (uchilaka) [inline package.json:49]: Don't need this here - there'll be a separate PR for updating the eslint config system

```suggestion
```

File: package.json [hunk 4]
```diff
@@ -40,9 +42,13 @@
...
+    "@playwright/test": "^1.42.1",
+    "@types/node": "^20.12.2",
```
Comment (uchilaka) [inline package.json:51]: Don't think I need this here

```suggestion
```

File: playwright.config.js [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+module.exports = defineConfig({
+  testDir: "./tests",
+  /* Run tests in files in parallel */
+  fullyParallel: true,
+  /* Fail the build on CI if you accidentally left test.only in the source code. */
+  forbidOnly: !!process.env.CI,
+  /* Retry on CI only */
+  retries: process.env.CI ? 2 : 0,
+  /* Opt out of parallel tests on CI. */
+  workers: process.env.CI ? 1 : undefined,
```
Comment (uchilaka) [inline playwright.config.js:23]: Anything above `1` worker was throwing errors so I stuck with just `1` for now

File: playwright.config.js [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+  },
+
+  /* Configure projects for major browsers */
+  projects: [
+    {
+      name: "chromium",
+      use: { ...devices["Desktop Chrome"] },
+    },
+
+    /* Test against Firefox. */
```
Comment (uchilaka) [inline playwright.config.js:42]: TODO: Try firefox & webkit again to see if they run successfully now 🤞🏾

File: .github/workflows/playwright.yml (hunk 1 above)
Comment (uchilaka) [reply .github/workflows/playwright.yml]: Confirmed: `v12.x` and `v14.x` are failing in CI. I'm getting an error related to the `ejs` package dependency: https://github.com/uchilaka/audit-export/actions/runs/8521632747/job/23340185623?pr=2

**Comment by uchilaka** [comment]
Playwright browser & node versions specs are passing: https://github.com/uchilaka/audit-export/actions/runs/8521836155

**Comment by TaylorHo** [comment]
Hey @uchilaka, thanks for the PR ❤️ 

I think that the test was not able to pass on versions lower than 16 because the format of the `npm audit` output has breaking changes between versions 14 and 16.

I'll review everything today and, if possible, include a test output of `npm audit` with some different NodeJS versions.

Again, I'm so thankful to you, a test suite will be a really great feature here! Thanks!
